cpp/tests/test_abm_world.cpp

```diff
@@ -299,52 +299,47 @@ TEST(TestWorld, evolveMigration)
+        // Create a person that is dead at time t - dt
+        auto& p_dead =
+            add_test_person(world, icu_id, mio::abm::AgeGroup::Age60to79, mio::abm::InfectionState::Dead, t - dt);
+        // Create a person that will be dead at time t
+        auto& p_severe =
+            add_test_person(world, home_id, mio::abm::AgeGroup::Age60to79, mio::abm::InfectionState::Dead, t);
         p_dead.set_assigned_location(icu_id);
-        p_dead.set_assigned_location(home_id);
-        p_dead.set_assigned_location(work_id);
-        p_critical.set_assigned_location(hospital_id);
-        p_critical.set_assigned_location(icu_id);
-
-        p_critical.add_new_infection(mio::abm::Infection(mio::abm::VirusVariant::Wildtype,
-                                                         mio::abm::AgeGroup::Age60to79, params, t + dt,
-                                                         mio::abm::InfectionState::Dead));
+        p_severe.set_assigned_location(home_id);
 
+        // Add trip to see if a dead person can move outside of cemetery by scheduled
         mio::abm::TripList& data = world.get_trip_list();
-        mio::abm::Trip trip1(p_dead.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(9), work_id, home_id);
-        mio::abm::Trip trip2(p_dead.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(10), home_id, icu_id);
-        mio::abm::Trip trip3(p_dead.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(10), home_id, icu_id);
-        mio::abm::Trip trip4(p_dead.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(20), work_id, home_id);
-        mio::abm::Trip trip5(p_dead.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(21), home_id, icu_id);
-        mio::abm::Trip trip6(p_dead.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(21), home_id, icu_id);
+        mio::abm::Trip trip1(p_dead.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(2), work_id, home_id);
+        mio::abm::Trip trip2(p_dead.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(3), home_id, icu_id);
+        mio::abm::Trip trip3(p_severe.get_person_id(), mio::abm::TimePoint(0) + mio::abm::hours(3), home_id, icu_id);
         data.add_trip(trip1);
         data.add_trip(trip2);
         data.add_trip(trip3);
-        data.add_trip(trip4);
-        data.add_trip(trip5);
-        data.add_trip(trip6);
 
-        // Check the dead person got burried and the critical person moved to ICU
-        world.evolve(t, dt);
+        // Check the dead person got burried and the servere person starts in Hospital
+        world.evolve(t - dt, dt);
         EXPECT_EQ(p_dead.get_location().get_type(), mio::abm::LocationType::Cemetery);
-        EXPECT_EQ(p_critical.get_location().get_type(), mio::abm::LocationType::ICU);
+        EXPECT_EQ(p_severe.get_infection_state(t - dt), mio::abm::InfectionState::InfectedSymptoms);
+        EXPECT_EQ(p_severe.get_location().get_type(), mio::abm::LocationType::Home);
```

Comment:
Why is the severely infected person not in hospital then?

Reply:
Yes, it should be at the hospital. I updated the code.